src/core/qgsellipsoidutils.cpp

```diff
@@ -186,12 +344,65 @@ QList<QgsEllipsoidUtils::EllipsoidDefinition> QgsEllipsoidUtils::definitions()
   sDefinitionCacheLock.unlock();
 
   sDefinitionCacheLock.lockForWrite();
+  QList<QgsEllipsoidUtils::EllipsoidDefinition> defs;
+
+#if PROJ_VERSION_MAJOR>=6
+  sEllipsoidCacheLock.lockForWrite();
+
+  PJ_CONTEXT *context = QgsProjContext::get();
+  PROJ_STRING_LIST authorities = proj_get_authorities_from_database( context );
```

Comment:
might potentially return nullptr in case of error (like proj.db not being found)